src/controllers/request-controller.ts

```diff
@@ -0,0 +1,126 @@
+import { OK, NOT_FOUND, BAD_REQUEST, CREATED } from "http-status-codes";
+import { Request as ExpReq, Response as ExpRes } from 'express';
+import { Logger, Logger as logger } from '@overnightjs/logger';
+
+import parser from 'cron-parser';
+import { config } from 'node-config-ts';
+
+import cors from 'cors';
+import { Controller, Get, Post, ClassMiddleware } from '@overnightjs/core';
+
+import CID from 'cids';
+import Context from '../context';
+import RequestService from '../services/request-service';
+import Contextual from '../contextual';
+import { RequestStatus } from '../models/request-status';
+import AnchorService from '../services/anchor-service';
+import { Anchor } from '../models/anchor';
+import Utils from "../utils";
+
+@Controller('api/v0/requests')
+@ClassMiddleware([cors()])
+export default class RequestController implements Contextual {
+  private anchorService: AnchorService;
+  private requestService: RequestService;
+
+  setContext(context: Context): void {
+    this.anchorService = context.lookup('AnchorService');
+    this.requestService = context.lookup('RequestService');
+  }
+
+  @Get(':cid')
+  private async get(req: ExpReq, res: ExpRes): Promise<ExpRes<any>> {
+    logger.Info(req.params.cid);
+
+    try {
+      const cid = new CID(req.params.cid);
+      const request = await this.requestService.findByCid(cid);
+      if (request == null) {
+        return res.status(NOT_FOUND).send({
+          error: 'Request doesn\'t exist',
+        });
+      }
+
+      switch (request.status) {
+        case RequestStatus.COMPLETED:
+          const anchor: Anchor = await this.anchorService.findByRequest(request);
+
+          return res.status(OK).json({
+            id: request.id,
+            status: RequestStatus[request.status],
+            cid: request.cid,
+            docId: request.docId,
+            createdAt: Utils.convertToUnixTimestamp(request.createdAt),
+            updatedAt: Utils.convertToUnixTimestamp(request.updatedAt),
+            anchorRecord: {
+              cid: anchor.cid,
+              content: {
+                path: anchor.path,
+                prev: anchor.request.cid,
+                proof: anchor.proofCid,
+              }
```

Comment:
The only thing that we really need to return here is the anchor CID. The ceramic-core node should given this CID be able to fetch the rest of the data as part of validating the proof.

Reply:
I've added `content` as an additional information so that the client doesn't need to fetch the data from IPFS. Caching purposes only. I can remove that and reformat the response. Opened ticket for that [14](https://github.com/ceramicnetwork/ceramic-anchor-service/issues/14).